src/components/config/index.ts

```diff
@@ -35,14 +41,26 @@ export const getConfig = () => {
   return cfg
 }
 
-export const updateConfig = (data: Config) => {
-  if (!cfg) {
-    cfg = {} as Config
+export async function* getConfigIterator() {
+  if (!cfg) throw new Error('Configuration setup has not been run yet')
+
+  yield cfg
+
+  if (!(process.env.CI || process.env.NODE_ENV === 'test')) {
+    yield* pEvent.iterator<string, Config>(emitter, CONFIG_UPDATED)
   }
+}
+
+export const updateConfig = (data: Partial<Config>) => {
+  const lastVersion = cfg.version
 
-  cfg.version = data.version
+  if (data.version) cfg.version = data.version
   if (data.probes) cfg.probes = data.probes
   if (data.notifications) cfg.notifications = data.notifications
+
+  if (cfg.version !== lastVersion) {
+    emitter.emit(CONFIG_UPDATED, cfg)
```

Comment:
Please consider wrapping event emitter, therefore all listener to the event can easily read.

Reply:
Currently this emitter is not exported and only used once. is it necessary to wrap the event emitter in this case?

Reply:
also pEvent.iterator method expect unwrapped event emitter as its argument

Reply:
Yes, you can do it later once the event emitter getting more complex.